[iOS] Cocoapods for iOS OSS release

xta0 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#25847 [iOS] Cocoapods for iOS OSS release**

### Summary

The Podspec file for iOS OSS release. This podspec contains the C++ header files and a static library that supports three architectures.

Please ignore the link for `s.source` for now, as I'm still working on the CI nightly build. This is a temporary link for testing purpose.

### Note

Previously I have a cocoapods release proposal  -  https://github.com/pytorch/pytorch/pull/25543 which contains two podspec files. However, for the time being, we haven't decided whether we want to release the Objective-C API wrapper or not. Please review and refer to this one if you have questions.

Differential Revision: [D17262459](https://our.internmc.facebook.com/intern/diff/D17262459)

xta0 committed:
[iOS] Cocoapods for iOS OSS release

Summary

Podspec file for iOS OSS release. This podspec contains the C++ header files and a static library that supports three architectures.

Note

Please ignore the link for `s.source` for now, as I'm still working on the CI nightly build. This is a temporary link for testing purpose.

xta0 committed:
Update on "[iOS] Cocoapods for iOS OSS release"


### Summary

The Podspec file for iOS OSS release. This podspec contains the C++ header files and a static library that supports three architectures.

Please ignore the link for `s.source` for now, as I'm still working on the CI nightly build. This is a temporary link for testing purpose.

### Note

Previously I have a cocoapods release proposal PR -  https://github.com/pytorch/pytorch/pull/25543 which contains two podspec files. However, for the time being, we haven't decided whether we want to release the Objective-C API wrapper or not. Please review and refer to this one if you have questions.

jspisak reviewed (approved):
Looks good @xta0 - thanks for making the naming changes!

xta0 committed:
Update on "[iOS] Cocoapods for iOS OSS release"


### Summary

The Podspec file for iOS OSS release. This podspec contains the C++ header files and a static library that supports three architectures.

Please ignore the link for `s.source` for now, as I'm still working on the CI nightly build. This is a temporary link for testing purpose.

### Note

Previously I have a cocoapods release proposal  -  https://github.com/pytorch/pytorch/pull/25543 which contains two podspec files. However, for the time being, we haven't decided whether we want to release the Objective-C API wrapper or not. Please review and refer to this one if you have questions.
